register_buildpack_test.go

```diff
@@ -0,0 +1,102 @@
+	"github.com/buildpacks/pack/internal/logging"
+	h "github.com/buildpacks/pack/testhelpers"
+
+	"github.com/heroku/color"
+	"github.com/sclevine/spec"
+	"github.com/sclevine/spec/report"
+)
+
+func TestRegisterBuildpack(t *testing.T) {
+	color.Disable(true)
+	defer color.Disable(false)
+	spec.Run(t, "rebase_factory", testRegisterBuildpack, spec.Parallel(), spec.Report(report.Terminal{}))
+}
+
+func testRegisterBuildpack(t *testing.T, when spec.G, it spec.S) {
+	when("#RegisterBuildpack", func() {
+		var (
+			fakeImageFetcher *ifakes.FakeImageFetcher
+			fakeAppImage     *fakes.Image
+			subject          *Client
+			out              bytes.Buffer
+		)
+
+		it.Before(func() {
+			fakeImageFetcher = ifakes.NewFakeImageFetcher()
+			fakeAppImage = fakes.NewImage("buildpack/image", "", &fakeIdentifier{name: "buildpack-image"})
+
+			h.AssertNil(t, fakeAppImage.SetLabel("io.buildpacks.buildpackage.metadata",
+				`{"id":"heroku/java-function","version":"1.1.1","stacks":[{"id":"heroku-18"},{"id":"io.buildpacks.stacks.bionic"},{"id":"org.cloudfoundry.stacks.cflinuxfs3"}]}`))
+			fakeImageFetcher.RemoteImages["buildpack/image"] = fakeAppImage
+
+			fakeLogger := logging.NewLogWithWriters(&out, &out)
+			subject = &Client{
+				logger:       fakeLogger,
+				imageFetcher: fakeImageFetcher,
+			}
+		})
+
+		it.After(func() {
+			_ = fakeAppImage.Cleanup()
+		})
+
+		when("registry type is github", func() {
+			const expectedURL = `https://github.com/jkutner/buildpack-registry/issues/new?body=%0A%23%23%23+Data%0A%0A%60%60%60toml%0Aid+%3D+%22heroku%2Fjava-function%22%0Aversion+%3D+%221.1.1%22%0Aaddr+%3D+%22buildpack-image%22%0A%60%60%60&title=ADD+heroku%2Fjava-function%401.1.1`
+			it("should open a github issue in the browser", func() {
+				switch runtime.GOOS {
+				case "linux":
+					execCommand = getFakeExecCommand(t, "xdg-open", expectedURL)
+				case "windows":
+					execCommand = getFakeExecCommand(t, "url.dll,FileProtocolHandler", expectedURL)
+				case "darwin":
+					execCommand = getFakeExecCommand(t, "open", expectedURL)
+				default:
+					// do nothing
+				}
+
+				h.AssertNil(t, subject.RegisterBuildpack(context.TODO(),
+					RegisterBuildpackOptions{
+						ImageName: "buildpack/image",
+						Type:      "github",
+						URL:       "https://github.com/jkutner/buildpack-registry",
+					}))
+			})
+
+			it("should throw error if missing URL", func() {
+				h.AssertError(t, subject.RegisterBuildpack(context.TODO(),
+					RegisterBuildpackOptions{
+						ImageName: "buildpack/image",
+						Type:      "github",
+						URL:       "",
+					}), "missing github URL")
+			})
+		})
+	})
+}
+
+func getFakeExecCommand(t *testing.T, expectedCmd, expectedURL string) func(command string, args ...string) *exec.Cmd {
+	return func(command string, args ...string) *exec.Cmd {
+		h.AssertEq(t, command, expectedCmd)
+		h.AssertEq(t, args[0], expectedURL)
+
+		cs := []string{"-test.run=TestHelperProcess", "--", command}
```

Comment:
For other reviewers: 
this is a super cool way of mocking out cmd calls in tests. Used in [os.exec_test](https://golang.org/src/os/exec/exec_test.go)

Reply:
Thanks, @dfreilich.  It was the best I can do, so I hope it's good enough.  I had to do some digging around to find this solution.  Can't take all the credit ;)